### ISSUE TITLE
Game freezes when idling more than 4 minutes. Flatpak AAGL 3.10.2

### ISSUE THREAD

**Issue by PopCat19** [an-anime-team/an-anime-game-launcher#403]
```
           .-------------------------:                    popcat19@popcat19-pc0
          .+=========================.                    ---------------------
         :++===++==================-       :++-           OS: CachyOS Linux x86_64
        :*++====+++++=============-        .==:           Host: 10181 (Lenovo K450e)
       -*+++=====+***++==========:                        Kernel: Linux 6.10.0-3-cachyos
      =*++++========------------:                         Uptime: 2 hours, 42 mins
     =*+++++=====-                     ...                Packages: 1368 (pacman), 3 (flatpak-system), 3 (flatpak-user)
   .+*+++++=-===:                    .=+++=:              Shell: fish 3.7.1
  :++++=====-==:                     -*****+              Display (MSI3CB3): 1920x1080 @ 165Hz *
 :++========-=.                      .=+**+.              Display (DELL S2415H): 1920x1080 @ 60Hz
.+==========-.                          .                 DE: KDE Plasma 6.1.3
 :+++++++====-                                .--==-.     WM: KWin (X11)
  :++==========.                             :+++++++:    WM Theme: CatppuccinMocha-Classic
   .-===========.                            =*****+*+    Theme: Breeze (CatppuccinMochaMauve) [Qt]
    .-===========:                           .+*****+:    Icons: Papirus-Dark [Qt], Papirus-Dark [GTK2/3/4]
      -=======++++:::::::::::::::::::::::::-:  .---:      Font: Rounded Mplus 1c (10pt, Medium) [Qt], Rounded Mplus 1c Medium]
       :======++++====+++******************=.             Cursor: breeze (24px)
        :=====+++==========++++++++++++++*-               Terminal: konsole 24.5.2
         .====++==============++++++++++*-                Terminal Font: Roboto Mono (10pt, Medium)
          .===+==================+++++++:                 CPU: Intel(R) Core(TM) i5-4460 (4) @ 3.40 GHz
           .-=======================+++:                  GPU: NVIDIA GeForce GTX 750 Ti [Discrete]
             ..........................                   Memory: 3.62 GiB / 15.55 GiB (23%)
                                                          Swap: 931.50 MiB / 23.00 GiB (4%)
                                                          Disk (/): 25.43 GiB / 238.42 GiB (11%) - btrfs
                                                          Disk (/home): 342.63 GiB / 673.34 GiB (51%) - ext4
                                                          Local IP (wlan0): 192.168.86.54/24
                                                          Locale: en_US.UTF-8
```

### Issue
I am using the Flatpak version of An Anime Game Launcher. The game freezes under 10 minutes of idling (usually 4 minutes) on `AAGL 3.10.2`. This issue only occurs when the game is idle (e.g. no interaction/unfocused window). So the game will still work fine when it's being played. This issue wasn't present until `AAGL 3.10.2`. 
|
The problem also occurs in character info and map screen from my testing. There's also a possible system instability after multiple game sessions (i.e. systemd-journald under memory pressure). 
|
This issue persists after a system reboot with the latest pacman and flatpak packages.

### Troubleshooting
I've ran multiple sessions for consistency, with low graphics preset and with the last working configuration (before 3.10.2): 
```
Wine-Staging-TKG 9.12, 
dxvk-gplasync-v2.3.1-1, 
window-mode exclusive (1024x768, in-game)
```

![Screenshot_20240720_234903](https://github.com/user-attachments/assets/19210d9e-1b65-42cc-80f4-c2ceafa2a6c7)

### Enviroment Tested
![image](https://github.com/user-attachments/assets/6454b024-2fb9-4c26-ba40-db34fb03e22a)
![image](https://github.com/user-attachments/assets/bf90f2e9-8dbb-423d-82f7-e9ea97731d6f)

### Btop: before freeze
![Screenshot_20240720_235056](https://github.com/user-attachments/assets/47edec8b-3788-4c75-bfb5-540366b9edaa)

### Btop: after freeze
`wine64` process status changes to `Sleeping`.
![Screenshot_20240720_235438](https://github.com/user-attachments/assets/c9d9dbb3-9860-40e0-bea2-19d11218b0ee)

### Logs
AAGL Info:
```
Anime Launcher SDK: 1.16.7
Anime Game Core: 1.21.3

gtk: 4.12.5
libadwaita: 1.4.5
pango: 1.51.2
cairo: 1.18.0
```

AAGL Config:
```
{
  "launcher": {
    "language": "en-us",
    "edition": "Global",
    "style": "Classic",
    "temp": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher",
    "repairer": {
      "threads": 4,
      "fast": false
    },
    "discord_rpc": {
      "app_id": 901534333360304168,
      "enabled": false,
      "title": "Researching the world",
      "subtitle": "of Teyvat",
      "icon": "launcher"
    },
    "environment": "PC",
    "behavior": "Nothing"
  },
  "game": {
    "path": {
      "global": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/Genshin",
      "china": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/YuanShen"
    },
    "voices": [],
    "wine": {
      "prefix": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/prefix",
      "builds": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/runners",
      "selected": "wine-9.12-staging-tkg-amd64",
      "sync": "FSync",
      "language": "System",
      "borderless": false,
      "drives": {
        "drive_c": true,
        "game_folder": "G",
        "map_folders": {}
      },
      "virtual_desktop": {
        "enabled": false,
        "width": 1920,
        "height": 1080
      },
      "shared_libraries": {
        "wine": true,
        "gstreamer": true
      }
    },
    "dxvk": {
      "builds": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/dxvks"
    },
    "enhancements": {
      "fsr": {
        "strength": 2,
        "quality": "Performance",
        "enabled": true
      },
      "gamemode": true,
      "hud": "DXVK",
      "fps_unlocker": {
        "path": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/fps-unlocker",
        "enabled": false,
        "config": {
          "fps": 165,
          "interval": 5000
        }
      },
      "gamescope": {
        "enabled": false,
        "game": {
          "width": 0,
          "height": 0
        },
        "gamescope": {
          "width": 0,
          "height": 0
        },
        "framerate": {
          "focused": 0,
          "unfocused": 0
        },
        "integer_scaling": true,
        "fsr": true,
        "nis": false,
        "window_type": "Borderless",
        "force_grab_cursor": false
      }
    },
    "environment": {},
    "command": null
  },
  "sandbox": {
    "enabled": false,
    "isolate_home": true,
    "hostname": null,
    "args": null,
    "private": [],
    "mounts": {
      "read_only": {},
      "bind": {},
      "symlinks": {}
    }
  },
  "components": {
    "path": "/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/components",
    "servers": [
      "https://github.com/an-anime-team/components"
    ]
  }
}

```

AAGL Debug:
[debug.log](https://github.com/user-attachments/files/16323113/debug.log)

Dmesg pruned (before reboot):
```
[ 9878.740779] audit: type=1326 audit(1721535199.812:4): auid=1000 uid=1000 gid=1000 ses=2 pid=116146 comm="wine-preloader" exe="/home/popcat19/.var/app/moe.launcher.an-anime-game-launcher/data/anime-game-launcher/runners/wine-9.12-staging-tkg-amd64/bin/wine-preloader" sig=31 arch=40000003 syscall=243 compat=1 ip=0x7d40000c code=0x0
[11140.619991] rtlwifi: AP off, try to reconnect now
[11140.620022] wlan0: Connection to AP d8:6c:63:f3:a9:48 lost
[11142.667983] systemd-journald[306]: Under memory pressure, flushing caches.
[11143.886975] wlan0: authenticate with d8:6c:63:f3:b8:3c (local address=c4:8e:8f:9f:45:8b)
[11143.886980] wlan0: send auth to d8:6c:63:f3:b8:3c (try 1/3)
[11143.987942] wlan0: send auth to d8:6c:63:f3:b8:3c (try 2/3)
[11144.091945] wlan0: send auth to d8:6c:63:f3:b8:3c (try 3/3)
[11144.195945] wlan0: authentication with d8:6c:63:f3:b8:3c timed out
[11144.715982] systemd-journald[306]: Under memory pressure, flushing caches.
[11144.914993] wlan0: authenticate with d8:6c:63:f3:a6:5c (local address=c4:8e:8f:9f:45:8b)
[11144.914999] wlan0: send auth to d8:6c:63:f3:a6:5c (try 1/3)
[11145.019949] wlan0: send auth to d8:6c:63:f3:a6:5c (try 2/3)
[11145.123955] wlan0: send auth to d8:6c:63:f3:a6:5c (try 3/3)
[11145.227961] wlan0: authentication with d8:6c:63:f3:a6:5c timed out
[11146.311472] wlan0: authenticate with d8:6c:63:f3:a9:4c (local address=c4:8e:8f:9f:45:8b)
[11146.311477] wlan0: send auth to d8:6c:63:f3:a9:4c (try 1/3)
[11146.411951] wlan0: send auth to d8:6c:63:f3:a9:4c (try 2/3)
[11146.515957] wlan0: send auth to d8:6c:63:f3:a9:4c (try 3/3)
[11146.619955] wlan0: authentication with d8:6c:63:f3:a9:4c timed out
[11160.186809] wlan0: authenticate with d8:6c:63:f3:a9:48 (local address=c4:8e:8f:9f:45:8b)
[11160.186815] wlan0: send auth to d8:6c:63:f3:a9:48 (try 1/3)
[11160.292015] wlan0: send auth to d8:6c:63:f3:a9:48 (try 2/3)
[11160.396016] wlan0: send auth to d8:6c:63:f3:a9:48 (try 3/3)
[11160.500017] wlan0: authentication with d8:6c:63:f3:a9:48 timed out
[11161.613057] systemd-journald[306]: Under memory pressure, flushing caches.
[11170.653783] wlan0: authenticate with d8:6c:63:f3:a9:48 (local address=c4:8e:8f:9f:45:8b)
[11170.653788] wlan0: send auth to d8:6c:63:f3:a9:48 (try 1/3)
[11170.656779] wlan0: authenticated
[11170.657058] wlan0: associate with d8:6c:63:f3:a9:48 (try 1/3)
[11170.667783] wlan0: RX AssocResp from d8:6c:63:f3:a9:48 (capab=0x1011 status=0 aid=3)
[11170.668197] wlan0: associated
[11170.675070] wlan0: Limiting TX power to 30 (30 - 0) dBm as advertised by d8:6c:63:f3:a9:48
```

**Comment by krypt0nn** [comment]
Probably will be fixed in https://github.com/an-anime-team/an-anime-game-launcher/pull/404

**Comment by PopCat19** [comment]
### Update
I've confirmed this issue only applies to AAGL version `3.10.2`.
|
I've temporarily rolled back to `AAGL 3.10.1` with:
```
flatpak update --commit=0e8dd669adfae6df804596e98f42816ed1a558f5c4f7ec428064d9c7733c54ec moe.launcher.an-anime-game-launcher
```
And ran the game for ~10 minutes with the same configuration as before.
|
The game runs as expected on versions before 3.10.2

<details>

<summary>btop difference</summary>

v3.10.1
![image](https://github.com/user-attachments/assets/c9235b82-aeb7-4e61-affc-3287c5ff0384)

v3.10.2
![Screenshot_20240720_235056](https://github.com/user-attachments/assets/47edec8b-3788-4c75-bfb5-540366b9edaa)

**Comment by nalsai** [comment]
The game froze almost immediately after becoming unfocused for me in 3.10.2, but it seems to be fixed in 3.10.3.

**Comment by PopCat19** [comment]
3.10.3 works, thank you
